test failed about testing Ispc Functions

hello,I meet a problem about test failed  about testing Ispc Functions
see:
```console
[root@centos8 tests]# ispc c-test-108.ispc
Warning: No output file or header file name specified. Program will be compiled and warnings/errors will be issued, but no output will be generated.
Warning: No --target specified on command-line. Using default system target "neon-i32x4".
Error: Invalid architecture "x86-64"
       Options: .
Error: arch = x86-64 and target = neon-i32x4 is not a valid combination.
[root@centos8 tests]# ispc --arch=aarch64 c-test-108.ispc
Warning: No output file or header file name specified. Program will be compiled and warnings/errors will be issued, but no output will be generated.
Warning: No --target specified on command-line. Using default system target "neon-i32x4".
Error: Invalid architecture "aarch64"
       Options: .
/tmp/root/spack-stage/spack-stage-ispc-1.14.1-qvpvty5hcf2lqk2oic4umtfsbmo23l5a/spack-src/src/ispc.cpp(1041): FATAL ERROR: Problem with target (neon-i32x4)
***
*** Please file a bug report at https://github.com/ispc/ispc/issues
*** (Including as much information as you can about how to reproduce this error).
*** You have apparently encountered a bug in the compiler that we'd like to fix!
***
/tmp/root/spack-stage/spack-stage-ispc-1.14.1-qvpvty5hcf2lqk2oic4umtfsbmo23l5a/spack-src/src/main.cpp(372): FATAL ERROR: Unhandled signal sent to process; terminating.
Aborted (core dumped)
```
Therefore, I want to know why the current test cases don't support the aarch64 platform?
### Information on your system
[root@centos8 tests]# spack debug report
* **Spack:** 0.16.1-1624-a0b5dcca3c
* **Python:** 3.6.8
* **Platform:** linux-centos8-aarch64
* **Concretizer:** original
<!-- Please include the output of `spack debug report` -->

<!-- If you have any relevant configuration detail (custom `packages.yaml` or `modules.yaml`, etc.) you can add that here as well. -->

@dbabokin how can I build ispc@1.15.0：
```console
[root@centos8 build]# cmake .. -DISPC_INCLUDE_TESTS=OFF
-- Build type not specified: Use Release by default.
-- Found LLVM LLVM_11_0
-- LLVM_CONFIG_EXECUTABLE: /home/spack/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-6y2fv2q5u65ltiri65e3rug6kadndva5/bin/llvm-config
-- CLANG_EXECUTABLE: /home/spack/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-6y2fv2q5u65ltiri65e3rug6kadndva5/bin/clang
-- CLANGPP_EXECUTABLE: /home/spack/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-6y2fv2q5u65ltiri65e3rug6kadndva5/bin/clang++
-- LLVM_DIS_EXECUTABLE: /home/spack/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-6y2fv2q5u65ltiri65e3rug6kadndva5/bin/llvm-dis
-- LLVM_AS_EXECUTABLE: /home/spack/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-6y2fv2q5u65ltiri65e3rug6kadndva5/bin/llvm-as
-- Detected LLVM version: 11.0.1
CMake Warning at cmake/LLVMConfig.cmake:134 (message):
  LLVM was built without assertions enabled (-DLLVM_ENABLE_ASSERTIONS=OFF).
  This disables dumps, which are required for ISPC to be fully functional.
Call Stack (most recent call first):
  CMakeLists.txt:174 (include)


-- GIT_BINARY: /usr/bin/git
fatal: not a git repository (or any of the parent directories): .git
-- M4 macro processor: /usr/bin/m4
-- Found Python3: /usr/bin/python3.8 (found version "3.8.3") found components: Interpreter
-- Found BISON: /usr/bin/bison (found suitable version "3.0.4", minimum required is "3.0")
-- Found FLEX: /usr/bin/flex (found suitable version "2.6.1", minimum required is "2.6")
llvm-config: unknown component name: arm
CMake Error at cmake/LLVMConfig.cmake:119 (message):
  Failed running
  /home/spack/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-6y2fv2q5u65ltiri65e3rug6kadndva5/bin/llvm-config;--libfiles;engine;ipo;bitreader;bitwriter;instrumentation;linker;option;frontendopenmp;arm;aarch64
Call Stack (most recent call first):
  cmake/LLVMConfig.cmake:156 (run_llvm_config)
  CMakeLists.txt:271 (get_llvm_libfiles)


-- Configuring incomplete, errors occurred!
See also "/tmp/root/spack-stage/spack-stage-ispc-1.15.0-zl7dxzngh5i2r4ww6z2ya2ne5ysv2i46/spack-src/build/CMakeFiles/CMakeOutput.log".

```

In the `centos8` Dockerfile on `aarch64` I have the following:
```
[root@a1156a156378 build]# bin/ispc stdlib_mask16_ispc.cpp
Warning: No output file or header file name specified. Program will be compiled and warnings/errors will be issued, but no output will be generated.
Warning: No --target specified on command-line. Using default system target "neon-i32x4".
stdlib_mask16_ispc.cpp:1:1: Error: syntax error, unexpected identifier.
namespace ispc {
^^^^^^^^^

```

Do you still have empty output from `--support-matrix`? I have the following:
```
[root@a1156a156378 build]# bin/ispc --support-matrix
                 | Windows | Linux        | Linux (custom) | FreeBSD      | macOS   | Android      | iOS     | PS4     | web
--------------------------------------------------------------------------------------------------------------------------------
sse2-i32x4       |         |              |                |              |         |              |         |         |
sse2-i32x8       |         |              |                |              |         |              |         |         |
sse4-i8x16       |         |              |                |              |         |              |         |         |
sse4-i16x8       |         |              |                |              |         |              |         |         |
sse4-i32x4       |         |              |                |              |         |              |         |         |
sse4-i32x8       |         |              |                |              |         |              |         |         |
avx1-i32x4       |         |              |                |              |         |              |         |         |
avx1-i32x8       |         |              |                |              |         |              |         |         |
avx1-i32x16      |         |              |                |              |         |              |         |         |
avx1-i64x4       |         |              |                |              |         |              |         |         |
avx2-i8x32       |         |              |                |              |         |              |         |         |
avx2-i16x16      |         |              |                |              |         |              |         |         |
avx2-i32x4       |         |              |                |              |         |              |         |         |
avx2-i32x8       |         |              |                |              |         |              |         |         |
avx2-i32x16      |         |              |                |              |         |              |         |         |
avx2-i64x4       |         |              |                |              |         |              |         |         |
avx512knl-i32x16 |         |              |                |              |         |              |         |         |
avx512skx-i32x8  |         |              |                |              |         |              |         |         |
avx512skx-i32x16 |         |              |                |              |         |              |         |         |
avx512skx-i8x64  |         |              |                |              |         |              |         |         |
avx512skx-i16x32 |         |              |                |              |         |              |         |         |
neon-i8x16       |         | arm, aarch64 | arm, aarch64   | arm, aarch64 |         | arm, aarch64 |         |         |
neon-i16x8       |         | arm, aarch64 | arm, aarch64   | arm, aarch64 |         | arm, aarch64 |         |         |
neon-i32x4       |         | arm, aarch64 | arm, aarch64   | arm, aarch64 |         | arm, aarch64 |         |         |
neon-i32x8       |         | arm, aarch64 | arm, aarch64   | arm, aarch64 |         | arm, aarch64 |         |         |
wasm-i32x4       |         |              |                |              |         |              |         |         |
genx-x8          |         |              |                |              |         |              |         |         |
genx-x16         |         |              |                |              |         |              |         |         |
```

@dbabokin ，If -DARM_ENABLED=TRUE is enabled, the following error message is displayed. What should I do?
```console
[root@localhost ~]# spack install --keep-stage -v ispc@main
……
[+] /home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-ekts5chfonko5mdjdfrlcw5qh3whwj3k
==> Installing ispc-main-hp6lpubhuy72qvidzz6ta7etpw5mfcs4
==> No binary for ispc-main-hp6lpubhuy72qvidzz6ta7etpw5mfcs4 found: installing from source
==> Warning: Missing a source id for ispc@main
==> No patches needed for ispc
==> ispc: Executing phase: 'cmake'
==> [2021-06-04-14:36:31.877262] 'cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/ispc-main-hp6lpubhuy72qvidzz6ta7etpw5mfcs4' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=OFF' '-DCMAKE_INSTALL_RPATH:STRING=/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/ispc-main-hp6lpubhuy72qvidzz6ta7etpw5mfcs4/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/ispc-main-hp6lpubhuy72qvidzz6ta7etpw5mfcs4/lib64;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-ekts5chfonko5mdjdfrlcw5qh3whwj3k/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/binutils-2.36.1-awuqsjdwsmimli3amjcak3intf2fhcyg/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/gettext-0.21-wspll52yyfbk2mmihjutav6777gcsvmb/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/bzip2-1.0.8-corln5kph65itvn6xbh3qdtx4aemcwmd/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/libiconv-1.16-dx2c372kkymixiwrw6jtba3gruhddu2d/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/libxml2-2.9.10-ufo5plslr5ucnhxrhv4a5iuyuzbh6vh3/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/xz-5.2.5-n6x7ibnlnr5sovxoy5kjhhglcjerbgyy/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/zlib-1.2.11-4jtfw6w6w3p7fk6kmewvik22samcjo6c/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/ncurses-6.2-nm2rk3htxjhi2sxfw2u27aeliihvuqsf/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/hwloc-2.4.1-yvcmhvpdovnr764xjyxutgcozlc6rwru/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/libpciaccess-0.16-t5wplngigwwetserfoefxgufneuvngcz/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/libedit-3.1-20191231-4awzo55npgcf3pgll5ghbhayfwrs35ip/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/pcre-8.44-yjg4oupkklquhpnpldj2r56uqdg3gfnd/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/z3-4.8.9-olcz3kwj244xbx66vpgzg5yhxi4mnlck/lib' '-DCMAKE_PREFIX_PATH:STRING=/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/cmake-3.20.3-ozavmkf2ztropx5a7or4flwmbkmyrmbv;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/python-3.8.8-jf6wr2bynimrvwful46fubsy53ikimhv;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/bison-3.7.4-fb3r6modpslbbvh7cjqlhgwlfykma2zq;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/flex-2.6.4-mdn4u3d7syumiruhemcruw35y7u7jvcg;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/ncurses-6.2-nm2rk3htxjhi2sxfw2u27aeliihvuqsf;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/zlib-1.2.11-4jtfw6w6w3p7fk6kmewvik22samcjo6c;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-ekts5chfonko5mdjdfrlcw5qh3whwj3k' '-DISPC_NO_DUMPS=ON' '/home/all_spack_env/spack_stage/root/spack-stage-ispc-main-hp6lpubhuy72qvidzz6ta7etpw5mfcs4/spack-src'
-- The C compiler identification is Clang 11.0.1
-- The CXX compiler identification is Clang 11.0.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-ekts5chfonko5mdjdfrlcw5qh3whwj3k/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-ekts5chfonko5mdjdfrlcw5qh3whwj3k/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found LLVM LLVM_11_0
-- LLVM_CONFIG_EXECUTABLE: /home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-ekts5chfonko5mdjdfrlcw5qh3whwj3k/bin/llvm-config
-- CLANG_EXECUTABLE: /home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-ekts5chfonko5mdjdfrlcw5qh3whwj3k/bin/clang
-- CLANGPP_EXECUTABLE: /home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-ekts5chfonko5mdjdfrlcw5qh3whwj3k/bin/clang++
-- LLVM_DIS_EXECUTABLE: /home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-ekts5chfonko5mdjdfrlcw5qh3whwj3k/bin/llvm-dis
-- LLVM_AS_EXECUTABLE: /home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-ekts5chfonko5mdjdfrlcw5qh3whwj3k/bin/llvm-as
CMake Error at cmake/FindLLVM.cmake:100 (message):
  Failed to find FileCheck
Call Stack (most recent call first):
  CMakeLists.txt:203 (include)


-- Configuring incomplete, errors occurred!
See also "/home/all_spack_env/spack_stage/root/spack-stage-ispc-main-hp6lpubhuy72qvidzz6ta7etpw5mfcs4/spack-build-hp6lpub/CMakeFiles/CMakeOutput.log".
==> Error: ProcessError: Command exited with status 1:
    'cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/ispc-main-hp6lpubhuy72qvidzz6ta7etpw5mfcs4' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=OFF' '-DCMAKE_INSTALL_RPATH:STRING=/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/ispc-main-hp6lpubhuy72qvidzz6ta7etpw5mfcs4/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/ispc-main-hp6lpubhuy72qvidzz6ta7etpw5mfcs4/lib64;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-ekts5chfonko5mdjdfrlcw5qh3whwj3k/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/binutils-2.36.1-awuqsjdwsmimli3amjcak3intf2fhcyg/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/gettext-0.21-wspll52yyfbk2mmihjutav6777gcsvmb/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/bzip2-1.0.8-corln5kph65itvn6xbh3qdtx4aemcwmd/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/libiconv-1.16-dx2c372kkymixiwrw6jtba3gruhddu2d/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/libxml2-2.9.10-ufo5plslr5ucnhxrhv4a5iuyuzbh6vh3/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/xz-5.2.5-n6x7ibnlnr5sovxoy5kjhhglcjerbgyy/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/zlib-1.2.11-4jtfw6w6w3p7fk6kmewvik22samcjo6c/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/ncurses-6.2-nm2rk3htxjhi2sxfw2u27aeliihvuqsf/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/hwloc-2.4.1-yvcmhvpdovnr764xjyxutgcozlc6rwru/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/libpciaccess-0.16-t5wplngigwwetserfoefxgufneuvngcz/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/libedit-3.1-20191231-4awzo55npgcf3pgll5ghbhayfwrs35ip/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/pcre-8.44-yjg4oupkklquhpnpldj2r56uqdg3gfnd/lib;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/z3-4.8.9-olcz3kwj244xbx66vpgzg5yhxi4mnlck/lib' '-DCMAKE_PREFIX_PATH:STRING=/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/cmake-3.20.3-ozavmkf2ztropx5a7or4flwmbkmyrmbv;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/python-3.8.8-jf6wr2bynimrvwful46fubsy53ikimhv;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/bison-3.7.4-fb3r6modpslbbvh7cjqlhgwlfykma2zq;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/flex-2.6.4-mdn4u3d7syumiruhemcruw35y7u7jvcg;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/ncurses-6.2-nm2rk3htxjhi2sxfw2u27aeliihvuqsf;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/zlib-1.2.11-4jtfw6w6w3p7fk6kmewvik22samcjo6c;/home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-ekts5chfonko5mdjdfrlcw5qh3whwj3k' '-DISPC_NO_DUMPS=ON' '/home/all_spack_env/spack_stage/root/spack-stage-ispc-main-hp6lpubhuy72qvidzz6ta7etpw5mfcs4/spack-src'

1 error found in build log:
     15    -- Found LLVM LLVM_11_0
     16    -- LLVM_CONFIG_EXECUTABLE: /home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-ekts5chfonko5mdjdfrlcw5qh3whwj3k/bin/llvm-config
     17    -- CLANG_EXECUTABLE: /home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-ekts5chfonko5mdjdfrlcw5qh3whwj3k/bin/clang
     18    -- CLANGPP_EXECUTABLE: /home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-ekts5chfonko5mdjdfrlcw5qh3whwj3k/bin/clang++
     19    -- LLVM_DIS_EXECUTABLE: /home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-ekts5chfonko5mdjdfrlcw5qh3whwj3k/bin/llvm-dis
     20    -- LLVM_AS_EXECUTABLE: /home/all_spack_env/spack/opt/spack/linux-centos8-aarch64/gcc-8.3.1/llvm-11.0.1-ekts5chfonko5mdjdfrlcw5qh3whwj3k/bin/llvm-as
  >> 21    CMake Error at cmake/FindLLVM.cmake:100 (message):
     22      Failed to find FileCheck
     23    Call Stack (most recent call first):
     24      CMakeLists.txt:203 (include)
     25
     26
     27    -- Configuring incomplete, errors occurred!

See build log for details:
  /home/all_spack_env/spack_stage/root/spack-stage-ispc-main-hp6lpubhuy72qvidzz6ta7etpw5mfcs4/spack-build-out.txt
```

--- a/.appveyor.yml
+++ b/.appveyor.yml
@@ -133,7 +133,7 @@ for:
   before_build:
     - sh: |-
         cd $ISPC_HOME && mkdir build && cd build
-        cmake ../ -DISPC_PREPARE_PACKAGE=ON -DCMAKE_INSTALL_PREFIX=$ISPC_HOME/install -DISPC_CROSS=ON -DWASM_ENABLED=$WASM
+        cmake ../ -DISPC_PREPARE_PACKAGE=ON -DCMAKE_INSTALL_PREFIX=$ISPC_HOME/install -DISPC_CROSS=ON -DWASM_ENABLED=$WASM -DCMAKE_CXX_FLAGS=-Werror
   build_script:
     - sh: make package -j4
 

--- a/.github/workflows/ispc-ci.yml
+++ b/.github/workflows/ispc-ci.yml
@@ -12,7 +12,7 @@ on:
         required: true
         default: 'full'
 env:
-  SDE_TAR_NAME: sde-external-8.59.0-2020-10-05
+  SDE_TAR_NAME: sde-external-8.63.0-2021-01-18
   LLVM_REPO: https://github.com/ispc/llvm-project
   TARGETS_SMOKE: '["avx2-i32x8"]'
   OPTSETS_SMOKE: "-O2"

--- a/.github/workflows/linux-nightly-trunk.yml
+++ b/.github/workflows/linux-nightly-trunk.yml
@@ -12,7 +12,7 @@ on:
   workflow_dispatch:
 
 env:
-  SDE_TAR_NAME: sde-external-8.59.0-2020-10-05
+  SDE_TAR_NAME: sde-external-8.63.0-2021-01-18
 
 jobs:
   # Building LLVM in docker, as using native Ubuntu 16.04 github-hosted image contains newer-than-expected libs and

--- a/.github/workflows/scripts/build-ispc.sh
+++ b/.github/workflows/scripts/build-ispc.sh
@@ -1,4 +1,4 @@
 #!/bin/bash -e
 echo PATH=$PATH
-cmake -B build -DISPC_PREPARE_PACKAGE=ON -DISPC_INCLUDE_BENCHMARKS=ON
+cmake -B build -DISPC_PREPARE_PACKAGE=ON -DISPC_INCLUDE_BENCHMARKS=ON -DCMAKE_CXX_FLAGS=-Werror
 cmake --build build --target package -j4

--- a/.travis.yml
+++ b/.travis.yml
@@ -74,14 +74,14 @@ jobs:
       script:
         # "Native-only" build (i.e. only arm and aarch64 are supported).
         - mkdir build-arm && cd build-arm
-        - cmake ..
+        - cmake .. -DCMAKE_CXX_FLAGS=-Werror
         - make -j32
         - make check-all
         - bin/ispc --support-matrix
         # Cross build (x86, x86-64, arm, aarch64 are supported, Linux only).
         - cd ..
         - mkdir build-all && cd build-all
-        - cmake .. -DX86_ENABLED=ON
+        - cmake .. -DX86_ENABLED=ON -DCMAKE_CXX_FLAGS=-Werror
         - make -j32
         - make check-all
         - bin/ispc --support-matrix

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -430,7 +430,7 @@ if (MSVC)
     set_source_files_properties(${FLEX_OUTPUT} PROPERTIES COMPILE_FLAGS "/wd4005 /wd4003")
     set_source_files_properties(${BISON_OUTPUT} PROPERTIES COMPILE_FLAGS "/wd4005 /wd4065")
 else()
-    target_compile_options(${PROJECT_NAME} PRIVATE -Wall -Wno-sign-compare -Wno-unused-function -Werror ${LLVM_CPP_FLAGS})
+    target_compile_options(${PROJECT_NAME} PRIVATE -Wall -Wno-sign-compare -Wno-unused-function ${LLVM_CPP_FLAGS})
     # The change implementing -Wno-unused-but-set-variable in clang was reverted, so commenting out for now.
     #if (${LLVM_VERSION_NUMBER} VERSION_GREATER_EQUAL "13.0.0")
     #    set_source_files_properties(${BISON_CPP_OUTPUT} PROPERTIES COMPILE_FLAGS "-Wno-unused-but-set-variable")

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -242,6 +242,10 @@ find_package(FLEX 2.6 REQUIRED)
 
 set (ISPC_MASKS 1 8 16 32 64)
 
+if (NOT X86_ENABLED AND NOT ARM_ENABLED AND NOT WASM_ENABLED AND NOT GENX_ENABLED)
+    message( FATAL_ERROR "Either X86, ARM, WASM or GENX targets need to be enabled.")
+endif ()
+
 if (X86_ENABLED)
     list(APPEND ISPC_TARGETS
         sse2-i32x4 sse2-i32x8

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -431,10 +431,6 @@ if (MSVC)
     set_source_files_properties(${BISON_OUTPUT} PROPERTIES COMPILE_FLAGS "/wd4005 /wd4065")
 else()
     target_compile_options(${PROJECT_NAME} PRIVATE -Wall -Wno-sign-compare -Wno-unused-function ${LLVM_CPP_FLAGS})
-    # The change implementing -Wno-unused-but-set-variable in clang was reverted, so commenting out for now.
-    #if (${LLVM_VERSION_NUMBER} VERSION_GREATER_EQUAL "13.0.0")
-    #    set_source_files_properties(${BISON_CPP_OUTPUT} PROPERTIES COMPILE_FLAGS "-Wno-unused-but-set-variable")
-    #endif()
     # Security options
     target_compile_options(${PROJECT_NAME} PRIVATE -fstack-protector-strong
                            -fdata-sections -ffunction-sections -fno-delete-null-pointer-checks

--- a/benchmarks/CMakeLists.txt
+++ b/benchmarks/CMakeLists.txt
@@ -59,6 +59,10 @@ set(BENCHMARK_ENABLE_TESTING OFF CACHE BOOL "Suppressing benchmark's tests" FORC
 set(BENCHMARK_ENABLE_INSTALL OFF CACHE BOOL "Suppressing benchmark's install" FORCE)
 add_subdirectory(vendor/google/benchmark)
 
+# Suppress warnings, as benchmarks are built with -Werror and trunk version of clang may trigger
+# new warnings, which are not yet fixed in Google Benchmark sources.
+target_compile_options(benchmark PRIVATE "-w")
+
 # Warn if build type is not release.
 if(CMAKE_BUILD_TYPE STREQUAL "Release")
     # NDEBUG may be excluded from build options globally, if LLVM build doesn't have that.

--- a/docker/ubuntu/cpu_ispc_build/Dockerfile
+++ b/docker/ubuntu/cpu_ispc_build/Dockerfile
@@ -61,7 +61,7 @@ RUN apt-get -y update && apt-get install -y m4 bison flex zlib1g-dev libc6-dev-i
 # Configure ISPC build
 RUN mkdir build_$LLVM_VERSION
 WORKDIR build_$LLVM_VERSION
-RUN cmake ../ -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=/usr/local/src/ispc/bin-$LLVM_VERSION
+RUN cmake ../ -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=/usr/local/src/ispc/bin-$LLVM_VERSION -DCMAKE_CXX_FLAGS=-Werror
 
 # Build ISPC
 RUN make ispc -j8 && make check-all && make install

